### PR TITLE
new makefile and updated headerfiles

### PR DESCRIPTION
**Pull request by JeanneXu1122**

### CODE REVIEW

File: include/map.h [hunk 1]
```diff
@@ -2,14 +2,14 @@
...
  */
 #ifndef MAP_H_
 #define MAP_H_
 #ifdef _WIN32
-#include <SDL.h>
-#include <sdl_ttf.h>
-#include <SDL_image.h>
+#include <SDL2/SDL.h>
+#include <SDL2/sdl_ttf.h>
+#include <SDL2/SDL_image.h>
```
Comment (layeghimahsa1996) [inline include/map.h:12]: I think we are not using sdl_image anymore. we can delete this line from here and also from makefile.

**Review by layeghimahsa1996 — APPROVED** [review]

File: Makefile [hunk 2]
```diff
@@ -1,106 +1,118 @@
...
 endif
 
-CC=gcc 
+#CC specifies which compiler we're using
+CC = gcc
+
+
 
-INCLUDES = include/map.h include/menu.h include/single_player.h include/map_textures.h include/multi_player.h
+INCLUDES = $(INCLUDE_PATHS) -Iinclude
```
Comment (Pari-Firoozi) [inline Makefile:24]: The following line should be there as these files are the prerequisites. 
INCLUDES = include/map.h include/menu.h include/single_player.h include/map_textures.h include/multi_player.h
If you remove that the .h files will not include at time of compilation.

The following line make it dependent to windows!
INCLUDES = $(INCLUDE_PATHS) -Iinclude

File: Makefile [hunk 3]
```diff
@@ -1,106 +1,118 @@
...
  		  build/single_player.o build/take_heart.o build/update_arrow.o build/update_barrier.o \
  		  build/update_player_pos.o build/user_manual.o build/win_game.o \
  		  build/options.o build/print_options.o build/multi_player.o build/generate_view_multi.o \
- 		  build/is_bullet_hit.o build/shoot.o build/update_bullet.o
+		  build/is_bullet_hit.o build/update_bullet.o build/shoot.o
  		 
-bin/main : run_tests $(OBJECTS)
-	$(CC) -o bin/main $(OBJECTS) $(LIBFLAGS)
+bin/main : $(OBJECTS)
+	$(CC) -o bin/main $(OBJECTS) $(LIBRARY_PATHS) $(COMPILER_FLAGS) $(LIBFLAGS)
```
Comment (Pari-Firoozi) [inline Makefile:34]: Make the Makefile dependent to windows.

File: Makefile [hunk 4]
```diff
@@ -1,106 +1,118 @@
...
+    
+#TARGET TO GENERATE THE OBJECT FILES FOR SRC.
+build/main.o: 
+	$(CC) -c src/main.c $(INCLUDES) -o build/main.o
 
-#TARGET TO GENERATE THE OBJECT FILES .
-build/main.o: src/main.c $(INCLUDES)
-	$(CC) -c src/main.c -o build/main.o
+build/credit.o:
+	$(CC) -c src/credit.c $(INCLUDES) -o build/credit.o
```
Comment (Pari-Firoozi) [inline Makefile:42]: INCLUDES should be in the prerequisites not receipts.

**Review by Pari-Firoozi — CHANGES_REQUESTED** [review]

File: include/single_player.h [hunk 5]
```diff
@@ -45,78 +43,23 @@ void game_over(SDL_Window* p_window);
...
  *
- * This function checks if a player is hit by arrows.
+ * This function checks if the player is hit by an arrow.
  * @param [in] map represent the map structure which has player position and arrows' positions.
  * @param [in] player_index  can be PLAYER1 or PLAYER2.
  * @return flag if the player is hit, flag = 1, otherwise flag = 0.
  */
 int is_player_hit(map_t* map, player_index_t player_index);
 
-/**
```
Comment (Pari-Firoozi) [inline include/single_player.h]: why you remove these comments? we need it to be in .h file for doxygen.

**Review by Pari-Firoozi — CHANGES_REQUESTED** [review]
I updated all comments for doxygen. Why you remove all?
I guess you add some other files by mistake in your commit.

**Review by Pari-Firoozi — COMMENTED** [review]
I have a suggestion about makefile. We can completely separate the makefile for windows. So you do not need to worry about other system. We put the whole thing for windows in if statement. And do the same for linux and mac. So they don't conflict because of different variables. So you Just write your make file for windows and put the whole thing including compilation of source and test files in that scope. I Think this is the best way to do that as windows does not know some other commands like rm, rf, echo,.... and you need to change them as well.

**Pull request closed without merging**